integrations/mistral/tests/test_mistral_chat_generator.py

```diff
@@ -267,3 +281,17 @@ def __call__(self, chunk: StreamingChunk) -> None:
 
         assert callback.counter > 1
         assert "Paris" in callback.responses
+
+    def test_live_run_with_tools(self, tools):
```

Comment:
A round trip test like [this](https://github.com/deepset-ai/haystack-core-integrations/blob/fd5def89af2a25041ae3117549d5ce410ec2c8af/integrations/google_ai/tests/generators/chat/test_chat_gemini.py#L362) would be more comprehensive: it tests tool call generation but also how the model behaves when you pass a tool call result.